src/arch/riscv/reg/riscv.rs

```diff
@@ -0,0 +1,70 @@
+//! RISC-V Register Definitions.
+//!
+//! Useful links:
+//!     - [GNU binutils-gdb XML Descriptions](https://github.com/bminor/binutils-gdb/blob/master/gdb/features/riscv)
+//!     - [riscv-tdep.h](https://github.com/bminor/binutils-gdb/blob/master/gdb/riscv-tdep.h)
+use crate::arch::Registers;
+use crate::internal::LeBytes;
+use num_traits::PrimInt;
+
+/// RISC-V Integer registers
+///
+/// The register width is set to `u32` or `u64` based on the `<U>` type.
+#[derive(Default)]
+pub struct RiscvCoreRegs<U> {
+    /// General purpose registers (x0-x31)
+    pub r: [U; 32],
```

Comment:
If the convention is to denote registers as `x0-x31`, then we might want to change this field from `pub r` to `pub x`.